src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Order/TreeItemOrderPropertyProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Order
+{
+    /// <summary>
+    /// Provider that computes display order of tree items based on input ordering of
+    /// evaluated includes from the project file.
+    /// </summary>
+    internal class TreeItemOrderPropertyProvider : IProjectTreePropertiesProvider
+    {
+        private readonly char[] _separators = new char[]
+        {
+            Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar
+        };
+
+        public IReadOnlyCollection<ProjectItemIdentity> OrderedItems { get; }
+
+        public ImmutableHashSet<string> AllowedItemTypes { get; }
+
+        public Dictionary<string, int> DisplayOrderMap { get; } = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+
+        public TreeItemOrderPropertyProvider(IReadOnlyCollection<ProjectItemIdentity> orderedItems)
```

Comment:
>TreeItemOrderPropertyProvider [](start = 15, length = 29)

Btw, i might be mistaken but in your repo convention is define ctors on top and properties below them?